### ISSUE TITLE
Fix python unit & integration tests

### ISSUE THREAD

**Issue by kevinschaper** [alliance-genome/agr#236]
Since they don't run automatically, they've drifted a little.

**Comment by kevinschaper** [comment]
It comes down to a mocking problem, or at least, any actual failures are masked by the mocking problem.  I think we're "[patching in the wrong place](http://alexmarandon.com/articles/python_mock_gotchas/)" - but I didn't manage to get the mock response into the right place.  

I did manage to get the .travis.yml set up to run both the python and node tests, and to cache pip & npm bits, so that it's still relatively quick.

**Comment by kevinschaper** [comment]
The work so far is here: https://github.com/alliance-genome/agr/tree/feature-python-unit-test-fixes

**Comment by oblodgett** [comment]
Just wondering about the "npm run build" does that need to be included in the travis file? I don't know why I am asking frontend questions.

**Comment by kevinschaper** [comment]
That's the part that failed on me when I just had travis calling "make install" - the node tests seem to pass after just doing "npm install" and that was what I'd seen in examples.   I could see doing a little bit of Makefile reorganizing to keep the travis config extra simple and clean though.

**Comment by kevinschaper** [comment]
I ended up putting "make build" back in, since it was necessary to build the webpack part.  It turned out to be failing because I wasn't getting the right node version by just specifying.  

All of the integration tests on travis will now fail with the same connection error, because there's no local Elasticsearch.  They're all attempting to mock the es.search function, so the connection error is a nice way to see that the mocking isn't working.  

I also added test_mock_es, which just tests whether the mock elasticsearch is called, and doesn't look at the content of any response, so that's the one to look at when running locally and trying to tell if mocking is working. 

I'll rename this issue to change the scope to just unit tests, and make a new issue for the mocking problem specifically.
